lib/postprocessing/gm-promisify.js

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+'use strict';
+
+const GM = require("gm").subClass({ imageMagick: !process.env.GM_USE_GRAPHICSMAGICK });
+const util = require("util");
+
+function wrapAndPromisify(obj, fnName) {
+    const originalFn = obj[fnName].bind(obj);
+    obj[fnName] = util.promisify(originalFn);
+}
+
+function gm(...arg) {
+    const img = GM(...arg);
+
+    // promisify gm output functions with callbacks
+    wrapAndPromisify(img, "write");
+    wrapAndPromisify(img, "stream");
+    wrapAndPromisify(img, "toBuffer");
+    wrapAndPromisify(img, "compare");
+    wrapAndPromisify(img, "thumb");
+    wrapAndPromisify(img, "format");
+    wrapAndPromisify(img, "identify");
```

Comment:
can we also add `wrapAndPromisify(img, "size");` to the promisified operations ?
pls see comment https://git.corp.adobe.com/nui/integration-tests/pull/77#discussion_r3075057

We could use `const GM = require('@adobe/asset-compute-sdk/lib/postprocessing/gm-promisify')` for reusability ?

Reply:
Will add it, but in a separate PR ok? Otherwise we have to wait until all the tests run again and we want to get this merged.

> We could use const GM = require('@adobe/asset-compute-sdk/lib/postprocessing/gm-promisify') for reusability ?

No, we should not do that. These deeper paths (`lib/`) really should stay an implementation detail of the module, so that we can change them without having to worry about breaking something.

As mentioned elsewhere, if we want to reuse this, we could have a small `gm-promisify` npm module for that.

Reply:
Turns out we need other changes anyway, so I will add it right away.